packages/dart_frog/lib/src/provider.dart

```diff
@@ -1,10 +1,90 @@
+import 'dart:async';
+
 import 'package:dart_frog/dart_frog.dart';
 
-/// Provide a value to the current handler by calling [create] lazily.
+final _cache = <Type, Object>{};
+
+/// Clears the in-memory provider cache.
+void clearProviderCache() => _cache.clear();
```

Comment:
Is there a reason this isn't exposed? Do we want to allow people to clear caches?

If so, it should have a generic type

Reply:
Is there a use case you have in mind?

Reply:
Clearing.. a cache of a provider 😅.

Let's say your API is a game and you the API level up through experience so it counts up until you hit a threshold, and then u reset the experience to 0 and show a "level up" message.

Bit of a convoluted example, but I dont see any direct downsides of exposing a way to clear a cache.

Reply:
Seems reasonable. Just to make sure I understand correctly, you're suggesting to expose something like:

```dart
/// Removes the cached value for the provided value of type [T].
void resetProvider<T>();
```

Do you feel it's still necessary to include an API to clear the entire cache?

Reply:
That would solve my use case!